foxpuppet/windows/browser/panel_ui/panel_ui.py

```diff
@@ -95,6 +95,11 @@ def open_new_window(self) -> None:
             )
             new_window = (set(self.selenium.window_handles) - initial_handles).pop()
             self.selenium.switch_to.window(new_window)
+            self.wait.until(
+                lambda _: self.selenium.execute_script("return document.readyState")
+                == "complete",
+                message="New window document not fully loaded",
+            )
```

Comment:
Is this needed?

Reply:
Yes, this more accurately addresses the root issue compared to `time.sleep`